jcli/src/jcli_lib/vote/tally/merge_results.rs

```diff
@@ -0,0 +1,274 @@
+use crate::jcli_lib::utils::io;
+use crate::jcli_lib::utils::OutputFormat;
+use jormungandr_lib::crypto::hash::Hash;
+use jormungandr_lib::interfaces::VotePlanId;
+use jormungandr_lib::interfaces::{PrivateTallyState, Tally, VotePlanStatus, VoteProposalStatus};
+use serde::Deserialize;
+use serde::Serialize;
+use std::collections::BTreeMap;
+use std::path::PathBuf;
+use structopt::StructOpt;
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("voteplan should be already decrypted before merging")]
+    VotePlanEncrypted,
+    #[error("voteplans have different privacy type")]
+    PrivacyMismatch,
+}
+
+#[derive(StructOpt)]
+#[structopt(rename_all = "kebab-case")]
+pub struct MergeVotePlan {
+    /// The path to json-encoded list of voteplans to merge. If this parameter is not specified, it
+    /// will be read from the standard input. Voteplans must be already decrypted before merging.
+    /// Two voteplans in the list will be merged if they have ALL the same proposals according to
+    /// the proposal (external) id.
+    #[structopt(long)]
+    vote_plans: Option<PathBuf>,
+    #[structopt(flatten)]
+    output_format: OutputFormat,
+}
+
+impl MergeVotePlan {
+    pub fn exec(&self) -> Result<(), super::Error> {
+        let voteplans: Vec<VotePlanStatus> =
+            serde_json::from_reader(io::open_file_read(&self.vote_plans)?)?;
+
+        let results = merge_voteplans(voteplans)?;
+
+        let output = self
+            .output_format
+            .format_json(serde_json::to_value(results)?)?;
+        println!("{}", output);
+
+        Ok(())
+    }
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq)]
+pub struct MergedVotePlan {
+    pub ids: [VotePlanId; 2],
+    pub proposals: Vec<VoteProposalStatus>,
+}
+
+fn merge_voteplans(voteplans: Vec<VotePlanStatus>) -> Result<Vec<MergedVotePlan>, Error> {
+    let mut results = Vec::with_capacity(voteplans.len() / 2);
+
+    for (curr, vp1) in voteplans.iter().enumerate() {
+        let p1: BTreeMap<Hash, VoteProposalStatus> = vp1
+            .proposals
+            .iter()
+            .map(|vps| (vps.proposal_id, vps.clone()))
+            .collect();
+
+        if let Some(vp2) = voteplans.iter().skip(curr + 1).find(|vp2| {
```

Comment:
why do we merge two voteplans only? Why not merging all of those with the same proposal set?